### ISSUE TITLE
Specify different type of UUID

### ISSUE THREAD

**Issue by steebchen** [prisma/prisma#2933]
Prisma currently only supports v4 UUIDs:

```prisma
model User {
  id   String  @id @default(uuid())
  test String?
}
```

However, it defaults to UUID v4, and there's no way to use v1 UUIDs. It's also unclear what it actually uses under the hood.

It would be best if the user could specify what version they want to use with a parameter, for example:

```
id String @id @default(uuid(v1))
```

Using uuid() without a parameter could then be deprecated, and existing users should replace it with `uuid(v4)`.

This would need to be supported by prisma schema, migrate and it would need to picked up by re-introspection.

more internal context: https://prisma-company.slack.com/archives/C1FPU5FPT/p1593790745201800

**Comment by albertoperdomo** [comment]
With the addition of a notation for DB-level defaults `dbgenerated(STRING)`, this could now be implemented with database functions.

Example:

```
model User {
  id   String  @id @db.Uuid @default(dbgenerated("uuid_generate_v1()"))
  test String?
}
```

Note that this allows for other types of UUID at the database-level, not at Prisma level.
This is currently in dev and schedule to be released in 2.16.0.

Do you think this is enough @steebchen or do you think we would also need these kinds of UUIDs as Prisma level defaults?

**Comment by steebchen** [comment]
seems good to me 👍

**Comment by albertoperdomo** [comment]
Update: This is now available in 2.16.0 via the nativeTypes preview feature. Release notes: https://github.com/prisma/prisma/releases/tag/2.16.0

**Comment by erdoganabaci** [comment]
first you have to use latest version @prisma/client and prisma then you can install extension pgcrypto in PostgreSql.Be ensure your crypto defination refer to public or any instance =>

`@id @default(dbgenerated("public.uuid_generate_v4()"))`

more details => 
https://stackoverflow.com/questions/35959265/postgresql-function-gen-random-uuid-not-working/66936848#66936848
